### DISCUSSION TITLE
How to add event to system set

### DISCUSSION THREAD

**Discussion by saberuster** [bevyengine/bevy#1774] [Q&A] [answered]
i want to add event like this:

```rust
    pub fn add_system_set_event<T>(&mut self) -> &mut Self
    where
        T: Component,
    {
        self.insert_resource(Events::<T>::default())
            .add_system_set_to_stage(CoreStage::First, SystemSet::new()
                .with_run_criteria(FixedTimestep::step(2.0))
                .with_system(Events::<T>::update_system.system()))
    }
```
because i want reuse the event system cache logic. 

i known timer system can do "fixed time" logic, but it is not cache.

I’m not sure if this example is a new requirement, or any other way can do this in bevy

**Comment by alice-i-cecile** [comment, marked as answer]
Ah: are you trying to ensure that your events persist across the fixed time step? Is there anything else you need from this functionality?

For now, I expect that swapping to manual event management (by replacing the update_system with your own logic) is the best approach.

**Reply by saberuster** [reply]
you are right，i will use my custom update_system to do this, thx.
